scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -89,6 +93,107 @@ class ScanCode(name: String, config: ScannerConfiguration) : LocalScanner(name,
         override fun create(config: ScannerConfiguration) = ScanCode(scannerName, config)
     }
 
+    /**
+     * The ScanCode JSON result file format.
+     */
+    data class Result(
+        // This old header format is used by Scancode version 2.x.
+        @JsonProperty("scancode_notice")
+        val notice: String?,
+        @JsonProperty("scancode_version")
+        val version: Semver?,
+        @JsonProperty("scancode_options")
+        val options: Map<String, Any>?,
+        val scanStart: String?, // Not yet available in version 2.2.1.
+        val filesCount: Int?,
+
+        // This new header format is used by Scancode version 3.x.
+        val headers: List<Header>?,
+
+        val files: List<FileResult>
+    ) {
+        /**
+         * The total count of scanned files.
+         */
+        val totalFilesCount = headers.orEmpty().fold(filesCount ?: 0) { count, header ->
+            count + header.extraData.filesCount
+        }
```

Comment:
If `filesCount` is not `null`, can there be headers with `extraData.filesCount`, or belong both to different ScanCode versions? Also, can there be multiple headers with `extraData.filesCount`? If this is about supporting different ScanCode versions I woudl prefer a more expressive implementation.